trestle/core/commands/author/ssp.py

```diff
@@ -0,0 +1,364 @@
+        control_dict: Dict[str, ControlHandle] = {}
+        for group in catalog.groups:
+            for control in group.controls:
+                control_handle = ControlHandle(group.id, group.title, control)
+                control_dict = self._get_controls(control_handle, control_dict)
+
+        # get list of control_ids needed by profile
+        control_ids: List[str] = []
+        for _import in profile.imports:
+            for include_control in _import.include_controls:
+                control_ids.extend(include_control.with_ids)
+
+        # get list of group id's and associated controls
+        needed_group_ids: Set[str] = set()
+        needed_controls: List[ControlHandle] = []
+        for control_id in control_ids:
+            control_handle = control_dict[control_id]
+            needed_group_ids.add(control_handle.group_id)
+            needed_controls.append(control_handle)
+
+        # make the directories for each group
+        for group_id in needed_group_ids:
+            (md_path / group_id).mkdir(exist_ok=True)
+
+        # assign values to class members for use when writing out the controls
+        self._param_dict = profile.modify.set_parameters
+        self._alters = profile.modify.alters
+        self._yaml_header = yaml_header
+        self._sections = sections
+
+        # write out the controls
+        for control_handle in needed_controls:
+            out_path = md_path / control_handle.group_id
+            self._write_control(out_path, control_handle)
+
+        return 0
+
+    def _get_implementations(self, control_file: pathlib.Path) -> List[ossp.ImplementedRequirement]:
+        control_id = control_file.stem
+        parse_tree = MarkdownValidator.load_markdown_parsetree(control_file)
+        tree = parse_tree[0] if len(parse_tree) == 1 else parse_tree[1]
+        imp_reqs: list[ossp.ImplementedRequirement] = []
+        ii = 0
+        while ii < len(tree):
+            item = tree[ii]
+            if item['type'] == 'thematic_break':
+                section: str = tree[ii + 1]['children'][0]['text']
+                if not section.startswith('Part '):
+                    break
+                text = tree[ii + 2]['children'][0]['text']
+                by_comp: ossp.ByComponent = gens.generate_sample_model(ossp.ByComponent)
+                by_comp.description = text
+                statement: ossp.Statement = gens.generate_sample_model(ossp.Statement)
+                part_label = section.split(' ')[1].replace('.', '')
+                statement_label = f'{control_id}_smt.{part_label}'
+                statement.by_components = {statement_label: by_comp}
+                imp_req: ossp.ImplementedRequirement = gens.generate_sample_model(ossp.ImplementedRequirement)
+                imp_req.control_id = control_id
+                imp_req.statements = {statement_label: statement}
+                imp_reqs.append(imp_req)
```

Comment:
So I am a little concerned here as to whether we have fully handled all scenarios. E.g. II cant' seee where we are searching for title objects and/or matching parts. This will expose us to bad behaviour.

I also think we might want to check a few scenarios w.r.t how we might handle things like:
- Multiple paragraphs of content
- Content that users put bolding or other features that might cause issues for this approach.

Reply:
Yes this is brittle.  I would like to revisit after 1.0.0 with a forward-looking design plan.

Reply:
Okay I've created this: https://github.com/IBM/compliance-trestle/issues/565 issue to track.